src/plan/barriers.rs

```diff
@@ -94,6 +94,20 @@ pub trait Barrier<VM: VMBinding>: 'static + Send + Downcast {
 
     /// Full post-barrier for array copy
     fn memory_region_copy_post(&mut self, _src: VM::VMMemorySlice, _dst: VM::VMMemorySlice) {}
+
+    /// A pre-barrier indicating that some fields of the object will probably be modified soon.
+    /// Specifically, the caller should ensure that:
+    ///     * The barrier must called before any field modification.
+    ///     * Some fields (unknown at the time of calling this barrier) might be modified soon, without a write barrier.
+    ///     * There are no safepoints between the barrier call and the field writes.
+    /// 
+    /// Currently, this is only used by the OpenJDK binding. The OpenJDK C2 slowpath allocation code
```

Comment:
Preferably use the OpenJDK case as an example, rather than a special case. You can decide whether this goes to `rustdoc` (with `///`), or just normal comments (with `//`). Either sounds okay to me.

Reply:
Done